app/models/restart_signal_handler.rb

```diff
@@ -1,11 +1,18 @@
 # JobQueue locks a mutex, hence the need for a separate SignalHandler thread
 # Self-pipe is also best practice, since signal handlers can themselves be interrupted
-class SignalHandler
+class RestartSignalHandler
+  class << self
+    alias listen new
```

Comment:
if we wanted to make it more testable I'd guess it should keep both the trap and the kill in some outside method ... 

`listen 'SIGUSR1' { Process.kill $$, 'SIGUSR2' }`